### PR TITLE
DDF-1264 created a new metadata_txt_ws field in Solr to be searched o…

### PR DESCRIPTION
**Pull request by wmcnalli**
…n for better wildcard matching

@kcwire
@millerw8
@pklinef
@wmcnalli
@jlcsmith
@beyelerb
@emanns95
@jrnorth
@coyotesqrl

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-catalog/105)

### CODE REVIEW

File: core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/TestSolrFilterDelegate.java [hunk 1]
```diff
@@ -152,6 +152,18 @@ public void testPropertyIsLike_AnyText_CaseSensitive() {
...
+    public void testPropertyIsLikeWildcard() {
+        String searchPhrase = "abc-123*";
+        String expectedQuery = SolrFilterDelegate.WHITESPACE_TOKENIZED_METADATA_FIELD + ":(abc\\-123*)";
+        boolean isCaseSensitive = false;
+
+        SolrQuery isLikeQuery = toTest.propertyIsLike(Metacard.ANY_TEXT, searchPhrase, isCaseSensitive);
+
+        assertThat(isLikeQuery.getQuery(), is(expectedQuery));
+
+    }
```
Comment (kcwire) [inline core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/TestSolrFilterDelegate.java]: Can we add more to this test? maybe searchPhrase "abc*" or "*123*" to make sure it's tokenizing text without hyphens correctly as well?

**Comment by coyotesqrl** [comment]
+1 to Keith's testing comment; otherwise, this looks good.

**Comment by coyotesqrl** [comment]
Pulled; built; tested. Verified.

**Comment by wmcnalli** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/TestSolrFilterDelegate.java, line 164 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/105#-Jsh0LYSVoiRq3a03g87-r1-164)** ([raw file](https://github.com/codice/ddf-catalog/blob/0b78c6ae0853f806663d1dd77cd0061575134bd4/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/TestSolrFilterDelegate.java#L164)):</sup>
OR'ing of filters is outside the scope of the FilterDelegate.  Search phrase "abc 123*" will be tokenized when it reaches Solr but the FilterDelegate will still return property:(abc 123*).

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/105)

**Comment by pklinef** [comment]
:+1:

**Comment by emanns95** [comment]
+1

**Comment by jrnorth** [comment]
LGTM

**Pull request merged**